common-rendering/src/components/keyEvents.tsx

```diff
@@ -0,0 +1,187 @@
+import { Pillar, Theme } from "@guardian/types";
+import { from } from "@guardian/src-foundations/mq";
+
+// ----- Component ----- //
+type paletteId = 300 | 400;
+
+interface KeyEvent {
+	time: string;
+	text: string;
+	url: string;
+}
+
+interface KeyEventsProps {
+	keyEvents: KeyEvent[];
+	theme: Theme;
+}
+
+interface ListItemProps {
+	keyEvent: KeyEvent;
+	theme: Theme;
+}
+
+const getColor = (theme: Theme, paletteId: paletteId) => {
+	switch (theme) {
+		case Pillar.Sport:
+			return palette.sport[paletteId];
+		case Pillar.Culture:
+			return palette.culture[paletteId];
+		case Pillar.Lifestyle:
+			return palette.lifestyle[paletteId];
+		case Pillar.Opinion:
+			return palette.opinion[paletteId];
+		default:
+			return palette.news[paletteId];
+	}
+};
+
+const keyEventWrapperStyles: SerializedStyles = css`
+	width: 100%;
+
+	${from.desktop} {
+		width: 13.75rem;
+	}
+`;
+
+const detailsStyles: SerializedStyles = css`
+	&:not([open]) .is-on,
+	&[open] .is-off {
+		display: none;
+	}
+
+	summary::-webkit-details-marker {
+		display: none;
+	}
+`;
+
+const titleRowStyles: SerializedStyles = css`
+	position: relative;
+	margin: ${remSpace[1]} ${remSpace[1]} 0 ${remSpace[3]};
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+
+	&:focus {
+		outline: none;
+	}
+
+	path {
+		fill: ${palette.neutral[46]};
+	}
+
+	svg {
+		height: 2rem;
+	}
+
+	${from.phablet} {
+		margin: ${remSpace[1]} ${remSpace[4]} 0;
+	}
+
+	${from.desktop} {
+		display: none;
+	}
+`;
+
+const titleStyle: SerializedStyles = css`
+	${headline.xxsmall({ fontWeight: "bold", lineHeight: "tight" })};
+	color: ${palette.neutral[7]};
+`;
+
+const listStyles: SerializedStyles = css`
+	margin: ${remSpace[3]};
+
+	${from.phablet} {
+		margin: ${remSpace[3]} ${remSpace[5]} 0;
+	}
+
+	${from.desktop} {
+		margin: ${remSpace[1]} 0 0;
+	}
+
+	li::before {
+		content: "";
+		border-color: transparent ${palette.neutral[7]};
+		border-style: solid;
+		border-width: 0.4rem 0 0.4rem 0.5rem;
+		display: block;
+		height: 0;
+		width: 0;
+		top: 0;
+		position: absolute;
+	}
+`;
+
+const listItemStyles: SerializedStyles = css`
+	padding-bottom: ${remSpace[3]};
+	border-left: 1px solid ${palette.neutral[7]};
+	position: relative;
+
+	&:last-child {
+		border-left: none;
+	}
+`;
+
+const timeTextWrapperStyles: SerializedStyles = css`
+	margin-left: ${remSpace[4]};
+`;
+
+const textStyles = (theme: Theme): SerializedStyles => css`
+	${headline.xxxsmall({ fontWeight: "regular", lineHeight: "regular" })};
+	color: ${getColor(theme, 300)};
+	text-decoration: none;
+`;
+
+const timeStyles: SerializedStyles = css`
+	${textSans.xxsmall({ fontWeight: "bold", lineHeight: "tight" })};
+	color: ${palette.neutral[7]};
+	display: block;
+`;
+
+const ListItem: FC<ListItemProps> = ({ keyEvent, theme }) => {
+	return (
+		<li css={listItemStyles}>
+			<div css={timeTextWrapperStyles}>
+				<time css={timeStyles}>{keyEvent.time}</time>
```

Comment:
```suggestion
				<time dateTime={keyEvent.time} css={timeStyles}>{keyEvent.time}</time>
```

The `time` tag requires the `datetime` attribute to be a valid, machine readable, element. The format of `keyEvent.time` must be a valid time string as defined here https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time

Reply:
Thanks @oliverlloyd 
So time tag would need a `datetime` attribute although we want to show the time like `2m ago`.

If the `keyEvent.time` is a valid date and then using timeAgo library we get the `<number>m ago`  from a function, could we use the `keyEvent.time` for `datetime` attribute and use the result of our function as a string as the content of `<time>{content}</time>`?

In any case, since we are having a seperate ticket to work on the timeAgo, is that OK to leave this as it is and update it in that PR?

Reply:
So yes, everything that you just said is spot on!


```
const displayString = timeAgo(new Date(myISOdate).getTime()); // timeAgo accepts epoch time

<time dateTime={myISOdate}>{displayString}</time>
```